src/algokit_utils/account_manager.py

```diff
@@ -0,0 +1,199 @@
+                                                 TransactionSigner)
+
+from .account import (get_dispenser_account, get_kmd_wallet_account,
+                      get_localnet_default_account)
+from .beta.client_manager import ClientManager
+
+
+@dataclass
+class AddressAndSigner:
+    address: str
+    signer: TransactionSigner
+
+class AccountManager:
+    """Creates and keeps track of signing accounts against sending addresses."""
+
+    def __init__(self, client_manager: ClientManager):
+        """
+        Create a new account creator.
+
+        :param client_manager: The ClientManager client to use for algod and kmd clients
+        """
+        self._client_manager = client_manager
+        self._accounts: Dict[str, TransactionSigner] = {}
+        self._default_signer: Optional[TransactionSigner] = None
+
+    def set_default_signer(self, signer: TransactionSigner) -> "AccountManager":
+        """
+        Sets the default signer to use if no other signer is specified.
+
+        :param signer: The signer to use, either a `TransactionSigner` or a `TransactionSignerAccount`
+        :return: The `AccountManager` so method calls can be chained
+        """
+        self._default_signer = signer
+        return self
+
+    def set_signer(self, sender: str, signer: TransactionSigner) -> "AccountManager":
+        """
+        Tracks the given account for later signing.
+
+        :param sender: The sender address to use this signer for
+        :param signer: The signer to sign transactions with for the given sender
+        :return: The AccountCreator instance for method chaining
+        """
+        self._accounts[sender] = signer
+        return self
+
+    def get_signer(self, sender: str) -> TransactionSigner:
+        """
+        Returns the `TransactionSigner` for the given sender address.
+
+        If no signer has been registered for that address then the default signer is used if registered.
+
+        :param sender: The sender address
+        :return: The `TransactionSigner` or throws an error if not found
+        """
+        signer = self._accounts.get(sender, None) or self._default_signer
+        if not signer:
+            raise ValueError(f"No signer found for address {sender}")
+        return signer
+
+
+    def get_information(self, sender: str) -> Dict[str, Any]:
+        """
+        Returns the given sender account's current status, balance and spendable amounts.
+
+        Example:
+            address = "XBYLS2E6YI6XXL5BWCAMOA4GTWHXWENZMX5UHXMRNWWUQ7BXCY5WC5TEPA"
+            account_info = account.get_information(address)
+
+        `Response data schema details <https://developer.algorand.org/docs/rest-apis/algod/#get-v2accountsaddress>`_
+
+        :param sender: The address of the sender/account to look up
+        :return: The account information
+        """
+        return cast(Dict[str, Any], self._client_manager.algod.account_info(sender))
```

Comment:
the typing of the algosdk is a bit frustrating, but for cases like this it is probably better to do something like this to narrow the type

```
info = self._client_manager.algod.account_info(sender)
assert isinstance(info, dict)
return info
```